docs/dev/local-dependencies.md

```diff
@@ -0,0 +1,286 @@
+`go.mod` and `go.sum` changes.
+
+If we want to use a specific tag instead of `HEAD` we can provide it in our
+calls:
+
+```
+go get github.com/openstack-k8s-operators/lib-common/modules/common@v0.1.0
+go get github.com/openstack-k8s-operators/lib-common/modules/database@v0.1.0
+go get github.com/openstack-k8s-operators/lib-common/modules/storage@v0.1.0
+```
+
+### Unmerged local simple reference
+
+There will be times when we'll be working on lib-common code in our local
+repository while using it in the cinder-operator and we want to test this.
+
+In this case we cannot use the `go get` because our lib-common module is not
+ within the cinder-operator command so we'll use the `go work edit` command
+instead:
+
+```
+go work edit -replace github.com/openstack-k8s-operators/lib-common/modules/common=../lib-common/modules/common
+go work edit -replace github.com/openstack-k8s-operators/lib-common/modules/database=../lib-common/modules/database
+go work edit -replace github.com/openstack-k8s-operators/lib-common/modules/storage=../lib-common/modules/storage
+```
+
+Alternatively we can use the `hack/setdeps.py` script to do the same thing:
+
+```
+hack/setdeps.py lib-common=../lib-common
+```
+
+Now we can proceed as usual with the [Building and
+Running](local.md#build-and-run) steps.  If we use `make build` then we'll need
+to let it know that it should use our workspace changes:
+
+```
+GOWORK= make build
+```
+
+Once we have confirmed that our code changes work as expected with the new
+dependency version we will need to submit the lib-common PR first and then the
+one in cinder-operator with the new dependency, as explained in the previous
+section.
+
+### Simple reference in PR
+
+Somebody may have submitted a PR to lib-common and we want to start writing
+code in the cinder-operator before that PR gets merged.
+
+Here we can either download the PR to our own local repository or just
+reference the one from GitHub.
+
+As an example, let's look at the `extraVolumes` effort and the [lib-common
+PR#88](https://github.com/openstack-k8s-operators/lib-common/pull/88).
+
+To reference it directly we'll go to GitHub's website and look for the source
+ of the PR, which in this case it's [fmount's extra_volumes branch](
+https://github.com/fmount/lib-common/tree/extra_volumes).
+
+So now we replace our lib-common with that one:
+
+```
+go work edit -replace github.com/openstack-k8s-operators/lib-common/modules/common=github.com/fmount/lib-common/modules/common@extra_volumes
+go work edit -replace github.com/openstack-k8s-operators/lib-common/modules/database=github.com/fmount/lib-common/modules/database@extra_volumes
+go work edit -replace github.com/openstack-k8s-operators/lib-common/modules/storage=github.com/fmount/lib-common/modules/storage@extra_volumes
+
+go mod tidy
+```
+
+Alternatively we can use the `hack/setdeps.py` to do all that for us:
+
+```
+hack/setdeps.py lib-common=88
```

Comment:
looking at he next like it seems 88 is the pull request ID, could it be explicitly mention?